pkg/internal/computeresources/transformer.go

```diff
@@ -100,24 +100,38 @@ func transformPodBasedOnLimitRange(p *corev1.Pod, limitRange *corev1.LimitRange)
 			p.Spec.Containers[i].Resources.Requests = defaultRequests
 		} else {
 			for _, name := range resourceNames {
-				setRequestsOrLimits(name, p.Spec.Containers[i].Resources.Requests, defaultRequests)
+				setRequests(name, p.Spec.Containers[i].Resources.Requests, defaultRequests, defaultContainerLimits)
 			}
 		}
 		if p.Spec.Containers[i].Resources.Limits == nil {
 			p.Spec.Containers[i].Resources.Limits = defaultContainerLimits
 		} else {
 			for _, name := range resourceNames {
-				setRequestsOrLimits(name, p.Spec.Containers[i].Resources.Limits, defaultContainerLimits)
+				setLimits(name, p.Spec.Containers[i].Resources.Limits, defaultContainerLimits)
 			}
 		}
 	}
 	return p
 }
 
-func setRequestsOrLimits(name corev1.ResourceName, dst, src corev1.ResourceList) {
-	if compare.IsZero(dst[name]) && !compare.IsZero(src[name]) {
-		dst[name] = src[name]
+// setRequests sets the requests to the max of (currentRequests, defaultRequests)
+// and the min of (currentRequests, defaultLimits)
+func setRequests(name corev1.ResourceName, dst corev1.ResourceList, requests corev1.ResourceList, limits corev1.ResourceList) {
+	if compare.IsZero(requests[name]) {
+		return
 	}
+	dst[name] = compare.MaxRequest(dst[name], requests[name])
+	if !compare.IsZero(limits[name]) {
+		dst[name] = compare.MinLimit(dst[name], limits[name])
+	}
+}
+
+// setLimits sets the limits to the min of (currentLimits, defaultLimits)
+func setLimits(name corev1.ResourceName, dst corev1.ResourceList, limits corev1.ResourceList) {
+	if compare.IsZero(limits[name]) {
+		return
+	}
+	dst[name] = compare.MinLimit(dst[name], limits[name])
```

Comment:
main changes related to this PR (just to exclude the depended changes in `tasklevel.go`)